tutorials/k8s_workload.md

```diff
@@ -0,0 +1,352 @@
+* `PUT /api/server/v1/virtualizationConnectors/{vcId}/securityGroups/{sgId}/members` with json payload:  
+**vcId**: "VIRTUALIZATION_CONNECTOR_ID"
+**sgId**: "SECURITY_GROUP_ID"
+```json
+{
+	"parentId": VIRTUALIZATION_CONNECTOR_ID,
+	"id": SECURITY_GROUP_ID,
+	"members": [
+	 {
+		"name": "env-production",
+		"type": "LABEL",
+		"label": "env=production"
+	  }
+	]
+}
+```
+
+On the OSC UI you should see the **Last Job Status**  of the created security group is **PASSED**. When clicking on **Membership** you should see one member in the security group with the IP address `70.70.0.2`.  
+
+
+### 8. Bind Security Group  
+Using the OSC api-doc `https://OSC_IP:8090/api-doc -> Operations for Virtualization Connectors`:  
+* `PUT /api/server/v1/virtualizationConnectors/{vcId}/securityGroups/{sgId}/bindings` with json payload:  
+**vcId**: `VIRTUALIZATION_CONNECTOR_ID`  
+**sgId**: `SECURITY_GROUP_ID`  
+```json
+	[
+	  {
+	    "virtualSystemId": "VIRTUAL_SYSTEM_ID",
+	    "name": "sgbind",
+	    "policyIds": [
+	      "POLICY_ID" # Retrieve from GET /api/server/v1/virtualSystems/{vsId}/policies
+	    ],
+	    "order": 0,
+	    "markedForDeletion": false,
+	    "binded": true
+	  }
+	]
+```
+
+After binding, ensure that the **Last Job Status** is **PASSED**.
+
+## Validate the Setup 
+
+#### VNF Deployment  
+
+After setting up OSC and deploying a **Distributed Appliance Instance** through the creation of a **Deployment Spec**, verify the **Distributed Appliance Instance** was deployed on Kubernetes: 
+```sh
+$  kubectl get deployments
+NAME        DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+testds-XX   1         1         1            1           3h
+```
+There should be one deployment prefixed with the name of the deployment spec created on OSC.  
+
+You can also see the corresponding pod:
+```sh
+$ kubectl get pods | grep testds
+testds-XX-XXXXXXXXX   1/1       Running
+```
+
+## Appendix
+
+### Network Redirection
+
+When using a real SDN controller plugin, like [Nuage](https://github.com/opensecuritycontroller/osc-nuage-plugin), you should be able to validate the traffic redirection when a policy is bound to a security group. For that you can perform the following steps on your Kubernetes cluster:  
+
+1. SSH to the node hosting `attacker-busybox` pod. To know which Kubernetes node is hosting the pod you can run:  
+```sh
+$ kubectl describe pods attacker-busybox | grep Node
+Node:           minion01
+```
+
+2. SSH to the node hosting the security VNF pod. To know which Kubernetes node is hosting the security VNF pod you can run:  
+```sh
+$ kubectl describe pods testds-XX-XXXXXXXXX  | grep Node  # Replace with the real pod name
+Node:           minion01
+```
+
+3. From the SSH session for the `attacker-busybox` identify the container ID (`ATTACKER_CONTAINER_ID`) and IP address (`ATTACKER_IP_ADD`) of  the `attacker-busybox`:  
+```sh
+$ docker ps | grep k8s_attacker-busybox
+```
+```sh
+$ docker exec ATTACKER_CONTAINER_ID ping ip add
```

Comment:
ping **ip add**?

Should this be ping IP_ADDR_OF_WORKLOAD 

looks like this is redundant and we should remove as its covered by step 4

Reply:
No, this should be just `ip add` to identify the IP of the attacker to be used in the `grep` below. The `ping` on this line was  a typo, fixing it.